src/OpenLoco/src/Windows/ObjectSelectionWindow.cpp

```diff
@@ -249,28 +265,74 @@ namespace OpenLoco::Ui::Windows::ObjectSelectionWindow
         }
 
         // Add a marker to denote the last tab
-        _tabInformation[tabPos].index = 0xFF;
+        _tabPositions[tabPos].index = 0xFF;
 
-        const auto firstTabIndex = ObjectType(_tabInformation[0].index);
+        const auto firstTabIndex = ObjectType(_tabPositions[0].index);
         repositionTargetTab(self, firstTabIndex);
     }
 
-    // 0x00472BBC
-    static ObjectManager::ObjIndexPair sub_472BBC(Window* self)
+    static bool contains(const std::string_view& a, const std::string_view& b)
     {
-        const auto objects = ObjectManager::getAvailableObjects(static_cast<ObjectType>(self->currentTab));
+        return std::search(a.begin(), a.end(), b.begin(), b.end(), [](char a, char b) {
+                   return tolower(a) == tolower(b);
+               })
+            != a.end();
+    }
 
+    static void applyFilterToObjectList()
+    {
+        std::string_view pattern = inputSession.buffer.c_str();
+        _numVisibleObjectsListed = 0;
+        for (auto& entry : _tabObjectList)
+        {
+            if (pattern.empty())
+            {
+                entry.visible = true;
+                _numVisibleObjectsListed++;
+                continue;
+            }
+
+            const std::string_view name = entry.object._name;
+            const std::string_view filename = entry.object._filename;
+
+            const bool containsName = contains(name, pattern);
+            const bool containsFileName = contains(filename, pattern);
+
+            entry.visible = containsName || containsFileName;
+
+            if (entry.visible)
+                _numVisibleObjectsListed++;
+        }
+    }
+
+    static void populateTabObjectList(ObjectType objectType)
+    {
+        _tabObjectList.clear();
+
+        const auto objects = ObjectManager::getAvailableObjects(objectType);
         for (auto [index, object] : objects)
         {
-            if (_50D144[index] & (1 << 0))
+            auto entry = TabObjectEntry{ index, object, true };
+            _tabObjectList.emplace_back(entry);
```

Comment:
emplace doesn't do anything here afaik. You would need to move into it for it to be worth it. As we know up front the size we want `_tabObjectList` to be you could resize it first on L312 then populate it after. `TabObjectEntry` is a pretty small class 3 pointers, 3 ints, 28 bytes doubt you gain much from emplaces.

Reply:
Ok, so just do `_tabObjectList.reserve(objects.size())` right?

What would you suggest using instead of `emplace_back`? Just insert or emplace?

Reply:
This is how you should do it with emplace_back.
https://godbolt.org/z/nbP5KzMac